### PR TITLE
[Enhancement] Support table level lock in Statement Planner

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+/**
+ * QueryLocker is used to obtain the lock corresponding to the metadata used in Query
+ * Will acquire the read locks of all tables involved in the query
+ * and the intention read locks of the database where the tables are located.
+ * <p>
+ * In terms of compatibility, when Config.use_lock_manager is turned off,
+ * it will be consistent with the original db-lock logic
+ * and obtain the db-read-lock of all dbs involved in the query.
+ */
+public class QueryLocker {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java:50]:
```suggestion
public class PlannerMetaLocker {
```

reason: not only used by query, and always used as a param of StatementPlanner.lock() to acquire lock on the meta being used. BTW, comments should also be updated.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+        Table table = metadataMgr.getTable(catalogName, dbName, tbName);
+        if (table == null) {
+            return null;
+        }
+
+        return new Pair<>(db, table);
+    }
+
+    private class TableCollector extends AstTraverser<Void, Void> {
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java]: You are right, but the current parameters in AnalyzeUtis are too complex. If you need to modify the parameters, you need to reconstruct a lot of calling logic. I don't want to reconstruct this logic in this PR. Moreover, Lock is also a function to be tested. Perhaps after sufficient testing, these logics can be unified. Currently it is only used by Plan Locker.

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    public void unlock() {
+        Locker locker = new Locker();
+        for (Map.Entry<Long, Set<Long>> entry : tables.entrySet()) {
+            Database database = dbs.get(entry.getKey());
+            List<Long> tableIds = new ArrayList<>(entry.getValue());
+            locker.unLockTablesWithIntensiveDbLock(database, tableIds, LockType.READ);
+        }
+    }
+
+    private Pair<Database, Table> resolveTable(ConnectContext session, TableName tableName) {
```
[the remark this answers is not in the document]
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryLocker.java]: same as below

**Review by nshangyiming — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by HangyuanLiu** [comment]
@Mergifyio backport branch-3.3

**Pull request merged**